### ISSUE TITLE
`command` missing from autin but present in `history | grep "command"`

### ISSUE THREAD

**Issue by weltonrodrigo** [atuinsh/atuin#1767]
I have a lot of commands starting with `kcat` but a CTRL-R `kcat` don't find them in atuin. When I do a `history | grep kcat` they show up.

If you need any further information about my environment I can provide.

One entry in history is :
```zsh
 5926  kcat -G spiaasdfs -t novo_saver \\n-b XXXXXXXXXXXX \\n-X security.protocol=sasl_ssl \\n-X sasl.mechanism=PLAIN \\n-X sasl.username='XXXXX' \\n-X sasl.password="XXXXXXXXX" -C -o begin -c 1000 > /tmp/XXXXXXX
```

**Comment by ahsandar** [comment]
I also see lot of commands are not in atuin history, initially I thought it was not adding commands that exited with an error but thats not the case. It just misses out on commands and its getting worse

**Comment by arcuru** [comment]
This is likely the oft-referenced bug with bash-preexec. The main branch has a potential fix if you can update to that. See #1750 

If you're not using bash and bash-preexec then please let us know the shell you're using

**Comment by weltonrodrigo** [comment]
I'm using zsh with oh-my-zsh. Was using fzf before atuin.

**Comment by arcuru** [comment]
1. Are you referring to commands you have in your history from before you started using Atuin? And if so, did you [import](https://docs.atuin.sh/reference/import/) the history from zsh?
2. How are you initializing atuin in your zshrc file?

**Comment by weltonrodrigo** [comment]
> * Are you referring to commands you have in your history from before you started using Atuin? And if so, did you [import](https://docs.atuin.sh/reference/import/) the history from zsh?

Yes. I followed the steps on https://docs.atuin.sh/#quickstart with `atuin import auto`

I believe I have older commands (as per history's command number).

> * How are you initializing atuin in your zshrc file?

Followed https://docs.atuin.sh/guide/installation/#shell-plugin with `eval "$(atuin init zsh)"`

**Comment by ellie** [comment]
> Yes. I followed the steps on https://docs.atuin.sh/#quickstart with atuin import auto

Did the output of running the import have the correct number of commands?

Sometimes, depending on your shell setup, it may not have found the correct file

You can specify this with `HISTFILE=./path/to/the/file atuin import zsh`

@ahsandar if you're using bash, @arcuru's earlier advice applies. If not, please open another issue with your setup + issue

**Comment by ahsandar** [comment]
yes I am on bash. The atuin import auto worked fine and whenever command was missed running it imported it fine. I uninstalled and built from source and since then there hasn't been an issue

**Comment by uzvg** [comment]
I met this trouble too, I can find my history through `history` command:

```shell
history | grep tiddlywiki
 1716  tiddlywiki --savewikifolder  ./backup/public "[all[tiddlers]ag[Journal]!tag[密码(Passwords)]!prefix[$:/state/]!prefix[$:/temp/]]"
```

but it cannot find in the atuin TUI.

**Comment by igorsol** [comment]
I installed atuin today.
Auto-imported bash history which was pretty small (about 600 lines).
Then imported larger history from the old backup file (about 30000 lines) - used `HISTFILE` trick to import from custom file location,
The `atuin history list` outputs everything - more than 30000 lines.
But with Ctrl-R fuzzy search I cannot find old commands - they just don't appear in the search results.
At the same time if I run:
```
atuin history list |fzf
```
I can find everything
